statsdaemon.go

```diff
@@ -269,12 +270,17 @@ func processGauges(buffer *bytes.Buffer, now int64) int64 {
 	for g, c := range gauges {
 		lastValue, ok := trackedGauges[g]
 
-		if ok && c == lastValue {
+		if ok && c == math.MaxUint64 && *deleteGauges {
 			continue
+		} else if ok && lastValue > 0 && c == math.MaxUint64 && *deleteGauges == false {
```

Comment:
I think this is functional, but the length of these if statements is getting hard to parse. Any ideas for simplifying? Typically idiomatic Go moves longer if/else if/else if/else into a switch block to simplify.... (my attempt below)

```go
currentValue := c
lastValue, hasLastValue := trackedGauges[g]
var hasChanged bool
if c != math.MaxUint64 {
    hasChanged = true
}
switch {
    case hasChanged:
        fmt.Fprintf(buffer, "%s %d %d\n", g, currentValue, now)
        trackedGauges[g] = c
        gauges[g] = math.MaxUint64
        num++
    case !hasChanged && hasLastValue &&  !*deleteGauges:
        fmt.Fprintf(buffer, "%s %d %d\n", g, lastValue, now)
        num++
    default:
        continue
}
```